shim/shim.c

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <poll.h>
+#include <assert.h>
+#include <stdarg.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <signal.h>
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <stdint.h>
+#include <inttypes.h>
+
+#define MAX_POLL_FDS 5
+#define BUF_SIZE 1024
+
+#define WINSIZE 11
+#define KILLCONTAINER 18
+
+/* globals */
+/* Pipe used for capturing signal occurence */
+int pipe_fd[2];
+
+void add_pollfd(struct pollfd *poll_fds, nfds_t *nfds, int fd,  short events) {
+	struct pollfd pfd = { 0 };
+
+	assert(*nfds < MAX_POLL_FDS);
```

Comment:
`assert()` or `err_exit()`? I guess this is an indication of an programming rather than a runtime error, but if the check fails, it could be useful to dump `poll_fds`.